crates/polars-utils/src/idx_vec.rs

```diff
@@ -148,3 +192,103 @@ impl AsMut<[IdxSize]> for IdxVec {
         unsafe { std::slice::from_raw_parts_mut(self.data_ptr_mut(), self.len) }
     }
 }
+
+impl PartialEq for IdxVec {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_slice() == other.as_slice()
+    }
+}
+
+impl FromIterator<IdxSize> for IdxVec {
+    fn from_iter<T: IntoIterator<Item = IdxSize>>(iter: T) -> Self {
+        let iter = iter.into_iter();
+        if iter.size_hint().0 <= 1 {
+            let mut new = IdxVec::new();
+            for v in iter {
+                new.push(v)
+            }
+            new
+        } else {
+            let v = iter.collect::<Vec<_>>();
+            v.into()
+        }
+    }
+}
+
+impl From<Vec<IdxSize>> for IdxVec {
+    fn from(value: Vec<IdxSize>) -> Self {
+        if value.capacity() <= 1 {
+            let mut new = IdxVec::new();
+            if let Some(v) = value.first() {
+                new.push(*v)
+            }
+            new
+        } else {
+            let mut me = std::mem::ManuallyDrop::new(value);
+            IdxVec {
+                data: me.as_mut_ptr(),
+                capacity: NonZeroUsize::new(me.capacity()).unwrap(),
+                len: me.len(),
+            }
+        }
+    }
+}
+
+impl From<&[IdxSize]> for IdxVec {
+    fn from(value: &[IdxSize]) -> Self {
+        if value.len() <= 1 {
+            let mut new = IdxVec::new();
+            if let Some(v) = value.first() {
+                new.push(*v)
+            }
+            new
+        } else {
+            value.to_vec().into()
+        }
+    }
+}
+
+impl From<IdxSize> for IdxVec {
```

Comment:
I don't like this implementation existing, it is error-prone (and also doesn't exist for `Vec`).

Reply:
Yes, was doubting of this one as well. The goal is a fast new for a single element where we don't do the reserve check.

Maybe `new_from_first`?